release/models/qos/openconfig-qos-interfaces.yang

```diff
@@ -362,13 +368,25 @@ submodule openconfig-qos-interfaces {
     leaf dropped-pkts {
       type oc-yang:counter64;
       description
-        "Number of packets dropped by the queue due to overrun";
+        "Number of packets dropped by the queue due to overrun, that is tail-drop or WRED drop";
     }
 
     leaf dropped-octets {
       type oc-yang:counter64;
       description
-        "Number of octets dropped by the queue due to overrun";
+        "Number of octets dropped by the queue due to overrun, that is tail-drop or WRED drop";
+    }
+
+    leaf ecn-marked-pkts {
+      type oc-yang:counter64;
+      description
+        "Number of packets with ECN codepoint re-marked to CE from ECT(0),ECT(1) in the queue";
```

Comment:
The description should be expanded to clarify if this only counts packets which the device changed the ECN marking?  Or does it also count packets which have existing ECN marking?    

In the Arista documentation it is clear that implementation counts BOTH packets that arrive with an ECN marking and also packets which the device has changed the CE bit from 0 to 1.    In the Cisco documentation it is not explicitly defined.  I did not check the other references.

Reply:
In my opinion, it seems more useful to count packets that are explicitly marked to CE by the device. This immediately identifies the device responsible for the ECN marking. The alternative of counting all CE packets egressing out of an interface mixes two different signals together:
1. Some prior device in the network marked the packet
2. The local device marked the packet.

Reply:
I tend to agree.  @rszarecki can you emphasize that this counter only counts packets which the device changed the ECN marking.  Perhaps:  

```
Number of packets which the system has changed the ECN codepoint to CE from ECT(0),ECT(1) in the queue.  Packets which are already marked CE and forwarded on this interface are not counted."
```

Reply:
After some thought I belive counting only packets that are re-marked form ECT(0), ECT(1) to CE is not best. It indeed allos to identify 1st congested interface in the path, but HIDES other congested interfaces along path. Hence if we use this counter as signal to "fix" congestion (e.g. add capacity), we would hit next congested interface.
If counter would count both a) remarked packets and b) incomming CE packets but ONLY when queue utilization is above treshold, then we get simultanous signal of all interfaces along path that can potentially remark ECN bits. Hence we would know all interfaces that nieed "fix" at once.

Reply:
Perhaps we need a separate counter for packets dropped due to RED/WRED, does this satisfy the use case?

There is not a RED/WRED packet drop counter today in OC, and I am not sure if this is commonly available from hardware, but if we create a new path for this, it would be a new path such as 
`/qos/interfaces/interface/output/scheduler/policy/schedulers/scheduler/state/red-drop-pkts`